tests/flows/validator-manager/migrate_from_v1.go

```diff
@@ -0,0 +1,697 @@
+	Expect(err).Should(BeNil())
+	utils.WaitForTransactionSuccess(context.Background(), l1AInfo, tx.Hash())
+
+	// Check that previous validator is still registered
+	// Migrate already registered validators to the new ValidatorManager version
+	validatorManager, err := validatormanager.NewValidatorManager(validatorManagerProxy.Address, l1AInfo.RPCClient)
+	Expect(err).Should(BeNil())
+
+	validationID, err := validatorManager.RegisteredValidators(&bind.CallOpts{}, poaNodeID)
+	Expect(err).Should(BeNil())
+	Expect(validationID[:]).Should(Equal(poaValidationID[:]))
+
+	// Check the subnetID was migrated
+	subnetID, err := validatorManager.SubnetID(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	Expect(subnetID[:]).Should(Equal(l1AInfo.SubnetID[:]))
+
+	//
+	// Attempt to remove the PoA validator before migrating it
+	//
+	posStakingManager, err := istakingmanager.NewIStakingManager(stakingManagerAddress, l1AInfo.RPCClient)
+	Expect(err).Should(BeNil())
+
+	opts, err = bind.NewKeyedTransactorWithChainID(ownerKey, l1AInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+	tx, err = posStakingManager.ForceInitiateValidatorRemoval(
+		opts,
+		validationID,
+		false,
+		0,
+	)
+	Expect(err.Error()).Should(ContainSubstring("reverted"))
+
+	//
+	// Migrate the validator
+	//
+
+	opts, err = bind.NewKeyedTransactorWithChainID(ownerKey, l1AInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+	tx, err = validatorManager.MigrateFromV1(opts, poaValidationID, 0)
+	Expect(err).Should(BeNil())
+	utils.WaitForTransactionSuccess(ctx, l1AInfo, tx.Hash())
+
+	//
+	// Remove the PoA validator and re-register as a PoS validator
+	//
+
+	utils.InitiateAndCompleteEndPoSValidation(
+		ctx,
+		signatureAggregator,
+		ownerKey,
+		l1AInfo,
+		pChainInfo,
+		posStakingManager,
+		stakingManagerAddress,
+		validatorManagerProxy.Address,
+		poaValidationID,
+		expiry,
+		nodes[0],
+		1,
+		false,
+		time.Time{},
+		network.GetPChainWallet(),
+		network.GetNetworkID(),
+	)
+
+	expiry2 := uint64(time.Now().Add(24 * time.Hour).Unix())
+	posValidationID := utils.InitiateAndCompleteNativeValidatorRegistration(
+		ctx,
+		signatureAggregator,
+		fundedKey,
+		l1AInfo,
+		pChainInfo,
+		nativeStakingManager,
+		stakingManagerAddress,
+		validatorManagerProxy.Address,
+		expiry2,
+		nodes[0],
+		network.GetPChainWallet(),
+		network.GetNetworkID(),
+	)
+	validatorStartTime := time.Now()
+
+	// Delist the PoS validator
+	utils.InitiateAndCompleteEndPoSValidation(
+		ctx,
+		signatureAggregator,
+		fundedKey,
+		l1AInfo,
+		pChainInfo,
+		posStakingManager,
+		stakingManagerAddress,
+		validatorManagerProxy.Address,
+		posValidationID,
+		expiry2,
+		nodes[0],
+		1,
+		true,
+		validatorStartTime,
+		network.GetPChainWallet(),
+		network.GetNetworkID(),
+	)
+}
+
+func convertSubnetPoAV1(
+	n *localnetwork.LocalNetwork,
+	ctx context.Context,
+	l1 interfaces.L1TestInfo,
+	weights []uint64,
+	senderKey *ecdsa.PrivateKey,
+) (proxy network.ProxyAddress,
```

Comment:
super nit: newline here for separation between the params and return values, as well as all return values at same indentation.